### ISSUE TITLE
Create process to manually re-queue failed or uninitiated product downloads.

### ISSUE THREAD

**Issue by sharkinsspatial** [NASA-IMPACT/hls-sentinel2-downloader-serverless#31]
Currently, if an initialization of the downloader lambda is throttled of fails, the product message is returned to the DLQ and is not marked as downloaded in the granules table.  If the message is not re-processed before the message retention period it is removed from the DLQ.  

We need a process to request re-queueing all non-downloaded granules for a date.  This should be a lambda which receives a date string and then queries the granules table for all granules for that date where downloaded is false and re-adds messages for those granules to the main queue.

**Comment by chuckwondo** [comment]
@sharkinsspatial, do you want some way to reconcile what's in the DLQ against what the proposed lambda function re-queues, so that we don't end up re-driving messages from the DLQ for files that the lambda function ended up re-queuing?

Alternatively, is there any reason we don't want to simply trigger a re-drive via the AWS Console or CLI? Is it because we want the granularity of being able to specify a particular date rather than everything that's currently in the DLQ?

**Comment by sharkinsspatial** [comment]
@chuckwondo Good point, we should probably only run this when both the queues are empty to avoid duplicate processing.  The issue we are handling with this use case specifically is messages which have fallen off the queue due to duration and need to be re-queued from the database.

**Comment by chuckwondo** [comment]
Gotcha. Thanks for clarifying. Do we want to add this to our `fix/esa_data_ecosystem` branch?

**Comment by chuckwondo** [comment]
Which date field should we use for selecting granules? beginposition, endposition, or ingestiondate?

**Comment by chuckwondo** [comment]
@sharkinsspatial, I'm going with ingestiondate, unless you indicate otherwise.

**Comment by chuckwondo** [comment]
@sharkinsspatial, how do you envision invoking this lambda? Do you expect to simply run it via the AWS CLI, or do you want some other means to do so?

**Comment by chuckwondo** [comment]
Perfect. I'm wrapping up unit tests this morning, then I'll add the lambda to the stack and try triggering it as part of my integration test stack from the CLI. I might not bother with adding integration tests, if the effort isn't worth it for this interim fix.

**Comment by chuckwondo** [comment]
Oh, one more question: shall I have the handler function spit out a JSON of the missing granules that were requeued? Perhaps something like so:

```json
{
  "ingestiondate": "<DATE>",
  "granules": [
    {
      "id": "<ID>",
      "filename": "<FILENAME>",
      "download_url": "<URL>"
    }
  ]
}
```

**Comment by chuckwondo** [comment]
@sharkinsspatial, here's the PR: https://github.com/NASA-IMPACT/hls-sentinel2-downloader-serverless/pull/35
